remote/gitea/gitea.go

```diff
@@ -229,15 +232,41 @@ func (c *client) Repos(u *model.User) ([]*model.Repo, error) {
 		return nil, err
 	}
 
-	all, _, err := client.ListMyRepos(gitea.ListReposOptions{})
-	if err != nil {
-		return repos, err
-	}
+	// Gitea SDK forces us to read repo list paginated.
```

Comment:
This should be marked as workaround so it can be rm if it got fixed upstream

adding a `// TODO: rm if ... got solved?

*this is just my personal suggestion* :)

Reply:
ok It look's like the whole setDefaults for ListOptions in the sdk do not work as they should ...

so you should have to itterate ... but I'll make a bugfix upstream ... just update to this version after it is patchend :)

Reply:
Didn't see any option in Gitea API for getting all repos in 1 shot so without Gitea API changes better to use iteration now. Not sure if Gitea should be brave to offer queries without limiting number of results to avoid overloads; I can see some preallocations for max possible number of results in

https://gitea.com/gitea/go-sdk/src/branch/master/gitea/repo.go#L134

and this may hurt to set limit to high. So looping in Woodpecker seem to be fine for now.